client/client-lib/src/ln/mod.rs

```diff
@@ -46,12 +45,12 @@ pub struct LnClient {
 }
 
 impl ClientModulePlugin for LnClient {
+    const KIND: &'static str = "ln";
     type Decoder = <LightningModule as ServerModulePlugin>::Decoder;
     type Module = LightningModule;
-    const MODULE_KEY: ModuleKey = MODULE_KEY_LN;
 
-    fn decoder(&self) -> &'static Self::Decoder {
-        &LightningModuleDecoder
+    fn decoder(&self) -> Self::Decoder {
```

Comment:
I think I can actually revert this change. Initially I thought I'll have to be passing instance_id into decoder, but this turned out to be misguided.

Reply:
I think the the worry about allocating is a bit premature. If there's a perf issue, we could always cache the decoders (they are not going to be changing over the course of runtime) if allocating every time is a problem. We actually have `DecoderRegistry` right in the consensus already.

Reply:
We've just landed the PR where all the decoders are stuffed right into `Database`, so I think we should make this slightly less constrained/confusing, as this Arc allocation really is not going to matter, and is done for all other typed->erased types anyway.